ext/pdo/tests/pdo_param_auto.phpt

```diff
@@ -0,0 +1,122 @@
+--TEST--
+PDO Common: PDO::PARAM_AUTO
+--EXTENSIONS--
+pdo
+--SKIPIF--
+<?php
+$dir = getenv('REDIR_TEST_DIR');
+if (false == $dir) die('skip no driver');
+require_once $dir . 'pdo_test.inc';
+PDOTest::skip();
+
+$tmp = PDOTest::get_temp_dir();
+if (!$tmp)
+    die("skip Can't create temporary file");
+
+$file = $tmp . DIRECTORY_SEPARATOR . 'pdoblob.tst';
+$fp = fopen($file, 'w');
+if (!$fp)
+    die("skip Can't create temporary file");
+
+if (4 != fwrite($fp, 'test')) {
+    die("skip Can't create temporary file");
+}
+fclose($fp);
+clearstatcache();
+
+if (!file_exists($file))
+    die("skip Can't create temporary file");
+
+unlink($file);
+?>
+--FILE--
+<?php
+if (getenv('REDIR_TEST_DIR') === false) putenv('REDIR_TEST_DIR='.__DIR__ . '/../../pdo/tests/');
+require_once getenv('REDIR_TEST_DIR') . 'pdo_test.inc';
+$db = PDOTest::factory();
+
+$null = '';
+switch ($db->getAttribute(PDO::ATTR_DRIVER_NAME)) {
+    case 'mysql':
+        $bitType = 'bit';
+        break;
+    case 'pgsql':
+    case 'firebird':
+        $bitType = 'boolean';
+        break;
+    case 'odbc':
+    case 'dblib':
+        $null = 'NULL';
+        $bitType = 'bit';
+        break;
+    default:
+        $bitType = 'int';
+}
+
+$db->exec(<<<SQL
+CREATE TABLE test(
+    f_int int NOT NULL,
+    f_bool {$bitType} NOT NULL,
+    f_null int {$null},
+    f_lob varchar(32) NOT NULL,
+    f_str varchar(32) NOT NULL
+)
+SQL);
+
+$file = PDOTest::get_temp_dir() . DIRECTORY_SEPARATOR . 'pdoblob.tst';
+file_put_contents($file, 'blob content');
+$fp = fopen($file, 'r');
+if (!$fp) {
+    printf("Cannot create test file '%s'\n", $file);
+    exit(1);
+}
+
+$stmt = $db->prepare('INSERT INTO test VALUES(?,?,?,?,?)');
+$stmt->bindValue(1, 123, PDO::PARAM_AUTO);
+$stmt->bindValue(2, true, PDO::PARAM_AUTO);
+$stmt->bindValue(3, null, PDO::PARAM_AUTO);
+$stmt->bindValue(4, $fp, PDO::PARAM_AUTO);
+$stmt->bindValue(5, 'test1', PDO::PARAM_AUTO);
```

Comment:
Can you explain how does this test verify that the feature works correctly? Can you add an unhappy path? As of now, I can't see any difference than if you had used PARAM_STR here.